lib/src/algorithms.dart

```diff
@@ -32,10 +32,10 @@ int _comparableBinarySearch/*<T extends Comparable<T>>*/(
 /// the objects.
 ///
 /// Returns -1 if [value] is not in the list by default.
-int binarySearch/*<T extends Comparable<T>>*/(
+int binarySearch/*<T>*/(
     List/*<T>*/ sortedList, /*=T*/ value, { int compare(/*=T*/ a, /*=T*/ b) }) {
   if (compare == null) {
-    return _comparableBinarySearch(sortedList, value);
+    return _comparableBinarySearch(sortedList as List<Comparable>, value as Comparable);
```

Comment:
`as List<Comparable>` will break here if the list doesn't have a reified runtime type of `Comparable`, which we can't count on. What we've done in [`PriorityQueue`](https://github.com/dart-lang/collection/blob/master/lib/src/priority_queue.dart#L146-L147) is to have the `compare()` function default to a version that does an internal `as` check before calling `compareTo()`. The simplest solution would be to do that here and just delete `_comparableBinarySearch()` entirely.

Also, we should more clearly document that this will throw a runtime error if `compare()` is omitted and `T` isn't a subtype of `Comparable<T>`.